File: .NET/VS2010TrainingKit/Labs/IntroToMEF/Source/Ex1/C#/End/ContosoAutomotive.Extensions/TailspinQuery.cs
// ----------------------------------------------------------------------------------
// Microsoft Developer & Platform Evangelism
// 
// Copyright (c) Microsoft Corporation. All rights reserved.
// 
// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
// ----------------------------------------------------------------------------------
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
// ----------------------------------------------------------------------------------

namespace ContosoAutomotive.Extensions
{
    using System.Collections.Generic;
    using System.ComponentModel.Composition;
    using System.Linq;
    using ContosoAutomotive.Common;

    [Export(typeof(ICarQuery))]
    public class TailspinQuery : CarQueryBase
    {
        public TailspinQuery()
        {
            this.Name = "Tailspin Motorsport";
            this.Description = "Life in the fast lane. You can't live without the best: Tailspin.";
            this.ImagePath = "Images/tailspin.jpg";
        }

        protected override IEnumerable<Car> RunQuery(IEnumerable<Car> cars)
        {
            var results = from c in cars
                          where c.Make == "Tailspin"
                            && c.Model == "Tempor"
                            && c.Price <= 40000
                            && c.Year >= 2005
                            && c.Transmission == Transmission.Automatic
                            && c.SatelliteRadio == true
                            && c.MoonRoof == true
                            && c.Mpg >= 25
                            && c.Mileage <= 10000
                            && c.Interior == InteriorType.Suede
                          select c;

            return results;
        }
    }
}

File: .NET/VS2010TrainingKit/Labs/IntroToMEF/Source/Assets/VB/ContosoAutomotive.Extensions/TailspinQuery.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Imports System.Collections.Generic
Imports System.Linq
Imports System.ComponentModel.Composition
Imports ContosoAutomotive.Common

<Export(GetType(ICarQuery))>
Public Class TailspinQuery
    Inherits CarQueryBase

    Public Sub New()
        Me.Name = "Tailspin Motorsport"
        Me.Description = "Life in the fast lane. You can't live without the best: Tailspin."
        Me.ImagePath = "Images/tailspin.jpg"
    End Sub

    Protected Overloads Overrides Function RunQuery(ByVal cars As IEnumerable(Of Car)) As IEnumerable(Of Car)
        Dim results = From c In cars _
                    Where c.Make = "Tailspin" _
                    AndAlso c.Model = "Tempor" _
                    AndAlso c.Price <= 40000 _
                    AndAlso c.Year >= 2005 _
                    AndAlso c.Transmission = Transmission.Automatic _
                    AndAlso c.SatelliteRadio = True _
                    AndAlso c.MoonRoof = True _
                    AndAlso c.MPG >= 25 _
                    AndAlso c.Mileage <= 10000 _
                    AndAlso c.Interior = InteriorType.Suede _
                    Select c

        Return results
    End Function
End Class
